corehq/apps/userreports/reports/filters/choice_providers.py

```diff
@@ -344,6 +345,14 @@ def display(loc):
                 return loc.display_name
         return [Choice(loc.location_id, display(loc)) for loc in locations]
 
+    def get_sorted_choices_for_values(self, choices, user):
```

Comment:
Hey @Charl1996 

Can you share why this is needed here considering we are already calling "accessible_to_user" in `get_choices_for_known_values` which gets after this function?

Reply:
@mkangia Good question.

This `get_choices_for_values` method will [add the missing values](https://github.com/dimagi/commcare-hq/blob/1d16f61a949627460c125904e8f0549e23e88fe9/corehq/apps/userreports/reports/filters/choice_providers.py#L90) which are not in the `choices` list; the `choices` var is a list containing the result of `get_choices_for_known_values` which in this case takes the user's location into account. In this instance `get_choices_for_known_values` will return a result which is location bound, but then because the inaccessible location is part of `values` and not `choices` it gets added to `choices`, effectively bypassing the location restriction. 

What I did was making sure the `get_choices_for_values` gets passed in the correct (already location-bound) location filters.